charts/victoria-metrics-distributed/values.yaml

```diff
@@ -4,91 +4,131 @@ nameOverride: "vm-distributed"
-availabilityZones:
-  - 
-    # -- Availability zone name
-    name: zone-eu-1
-    # -- Allow data ingestion to this zone
-    allowIngest: true
-    # -- Allow data query from this zone through global query endpoint
-    allowQuery: true
-    # -- Node selector to restrict where pods of this zone can be placed.
-    # usually provided by cloud providers.
-    nodeSelector:
-      topology.kubernetes.io/zone: zone-eu-1
-    # -- Extra affinity adds user defined custom affinity rules
-    extraAffinity: {}
-      # preferredDuringSchedulingIgnoredDuringExecution:
-      # - weight: 1
-      #   preference:
-      #     matchExpressions:
-      #     - key: topology.kubernetes.io/zone
-      #       operator: In
-      #       values:
-      #       - us-central1-b
-    # -- Topology spread constraints allows to customize the default topologySpreadConstraints.
-    topologySpreadConstraints:
-      - maxSkew: 1
-        topologyKey: kubernetes.io/hostname
-        whenUnsatisfiable: ScheduleAnyway
-    vmauthIngest:
-      # -- Create vmauth as a local write endpoint
+write:
+  global:
+    vmauth:
+      # -- Create a vmauth as the global write entrypoint
       enabled: true
       # -- Override the name of the vmauth object
       name: ""
-      # -- VMAuth spec. More options can be found [here](https://docs.victoriametrics.com/operator/api/#vmauthspec)
+      # -- Spec for VMAuth CRD, see [here](https://docs.victoriametrics.com/operator/api#vmauthspec)
       spec:
-        extraArgs:
-          # vmauth discovers backend IPs via periodic DNS queries,
-          # and performs load balancing between vminsert instances
-          # check https://docs.victoriametrics.com/vmauth/#discovering-backend-ips
-          discoverBackendIPs: "true"
-    vmauthQueryPerZone:
-      # -- Create vmauth as a local read endpoint
+        port: "8427"
+
+read:
+  global:
+    vmauth:
+      # -- Create vmauth as the global read entrypoint
       enabled: true
       # -- Override the name of the vmauth object
       name: ""
-      # -- VMAuth spec. More options can be found [here](https://docs.victoriametrics.com/operator/api/#vmauthspec)
+      # -- Spec for VMAuth CRD, see [here](https://docs.victoriametrics.com/operator/api#vmauthspec)
       spec:
-        extraArgs:
-          discoverBackendIPs: "true"
-    vmauthCrossAZQuery:
-      # -- Create a vmauth with all the zone with `allowQuery: true` as query backends
-      enabled: true
-      # -- Override the name of the vmauth object
-      name: ""
-      # -- VMAuth spec. More options can be found [here](https://docs.victoriametrics.com/operator/api/#vmauthspec)
-      spec: {}
-    # -- VMAgent here only meant to proxy write requests to each az,
+        port: "8427"
+
+# Config per availability zone components, including vmagent, vmcluster, vmauth etc
+availabilityZones:
+  -
+    # -- Availability zone name
+    name: zone-eu-1
+    common:
```

Comment:
I think components shouldn't share other `spec` except nodeSelector/affinitytopologySpreadConstraints, which could lead to misuse and it's better to not lead to that way.
How about changing this as a plain `commonSpec` instead of having `common.spec`?

Reply:
`common.spec` allows having then `common.annotations` and `common.labels`